tests/test_session.py

```diff
@@ -43,12 +63,10 @@ def test_diff(self):
         self.assertEqual(diff, self.session.diff("import os"))
 
     def test_diff_file(self):
-        with tempfile.NamedTemporaryFile(mode="w", suffix=".py") as tmp:
-            tmp.write("import os")
-            tmp.seek(0)
-            diff_file = self.session.diff_file(path=Path(tmp.name))
+        with reopenable_temp_file("import os") as tmp_path:
+            diff_file = self.session.diff_file(path=tmp_path)
             diff = (
-                f"--- {tmp.name}\n",
+                f"--- {str(tmp_path)}\n",
```

Comment:
```suggestion
                f"--- {tmp_path.as_posix()}\n",
```

Reply:
This way, It won't work well with Windows!!!

Reply:
Thx for the suggestion.

Reply:
It is not specified in the document but it works, it can be contributed by adding the "PurePosixPath" sample to the document. In fact, there should be at least two system instances under each method.

https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.as_posix

Reply:
I see, you know what?
There's no diffrent between `str(tmp_path)` and `tmp_path.as_posix()` on our case!

Anyway, I've resolved it.

Reply:
Yes I know, the method I suggested is better.